### PR TITLE
Riegeli: add support for zstd compression

### PR DESCRIPTION
**Pull request by schroederc**

### CODE REVIEW

File: kythe/go/util/riegeli/golden_test.go [hunk 1]
```diff
@@ -89,11 +90,12 @@ func (j *jsonReader) Next() (*spb.Entry, error) {
...
 	for _, variant := range goldenRiegeliFileVariants {
 		file := strings.Join([]string{goldenRiegeliFilePrefix, variant, "riegeli"}, ".")
-		t.Run(variant, func(t *testing.T) { checkGoldenData(t, file) })
+		opts := strings.Replace(variant, "_", ",", -1)
+		t.Run(variant, func(t *testing.T) { checkGoldenData(t, file, opts) })
 	}
 }
 
-func checkGoldenData(t *testing.T, goldenRiegeliFile string) {
+func checkGoldenData(t *testing.T, goldenRiegeliFile, expectedOptions string) {
```
Comment (creachadair) [inline kythe/go/util/riegeli/golden_test.go]: Please document the format of the options string (e.g., is it flags? Space-separated? Other?)

File: kythe/go/util/riegeli/riegeli.go [hunk 2]
```diff
@@ -66,6 +66,16 @@ func BrotliCompression(level int) CompressionType {
 	return &compressionLevel{brotliCompression, level}
 }
 
+// ZSTDCompression returns a CompressionType for zstd compression with the
+// given compression level.  If level < 0 || level > 22, then the DefaultZSTDLevel
```
Comment (creachadair) [inline kythe/go/util/riegeli/riegeli.go]: What's special about 22?

File: kythe/go/util/riegeli/golden_test.go (hunk 1 above)
Comment (schroederc) [reply kythe/go/util/riegeli/golden_test.go]: It's the magic C++ standard (i.e. CSV of various options).

File: kythe/go/util/riegeli/riegeli.go (hunk 2 above)
Comment (schroederc) [reply kythe/go/util/riegeli/riegeli.go]: Dunno.  It's just part of the zstd spec.

**Pull request merged**